assets/js/phoenix/longpoll.js

```diff
@@ -95,7 +95,10 @@ export default class LongPoll {
           this.onerror(500)
           this.closeAndRetry(1011, "internal server error", 500)
           break
-        default: throw new Error(`unhandled poll status ${status}`)
+        default:
+          this.onerror(status)
+          this.close()
+          break
```

Comment:
This patch was changed a little. Here is how it was applied to 1.6.5: https://github.com/salemove/phoenix/commit/c11e6adf8e051bcd2688f20a0c6aa2dba9c0dd43

There were three changes:
1. Status is now emitted as a number directly instead of being wrapped inside an object. I changed this because the new Phoenix version calls `onerror` with the status code and I think we don't want to diverge from them too much, otherwise, it could lead to unexpected issues in the future.
2. The close reason is now set to 1008 in case of 403 instead of 1000.
3. The close is considered no longer clean in case of 403.

The first change requires some modifications in visitor-js-api and channel.

Reply:
This change also requires a few changes in visitor-js-api and channel

Reply:
> With the previous version, we did not reconnect if 403 was returned, now we do.

I don't remember exactly what the impact is with this. Can it actually recover by reconnecting if the token is expired or invalid? We don't want to create unnecessary noise and load on the system if we don't have to. It might be fine, it's been quite some time I've touched this part of the system.

Reply:
Actually, I meant to remove that last statement but I somehow failed to do so 😄 

When analysing the code, it seemed like we should not reconnect in case of 403 due to how the connection was being closed but I missed something and we do reconnect. There is one small change in how the connection is changed --- now it closes with `closeWasClean` set to `false` and status code set to 1008 (instead of 1000). While it can affect how some pathways in the code are taken, the end result should pretty much stay the same --- we reconnect in case of 403.